src/libs/core/src/data.cpp

```diff
@@ -2431,6 +2431,9 @@ DATA *DATA::GetVarPointer()
     // if(!bRef) return this;
     if (Data_type != VAR_REFERENCE)
         return this;
+    if (pReference == this) {
+        return this;
```

Comment:
If this is abnormal behaviour maybe we also want to add something like this?
```suggestion
        Error("ref points to itself");
        return this;
```

Reply:
Good idea